File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/GenericResponse.ts
import { Response, IResponse} from 'shared/AppModels/Response';
/**
 * Interface for: ProjectStart.Common.GenericResponse<T>
 */
export interface IGenericResponse<T> extends IResponse {
    data: T;
}
export class GenericResponse<T> extends Response implements IGenericResponse<T> {
    data: T;
}

File: ProjectStart/ProjectStart.Common/GenericResponse.cs
using System.Collections.Generic;
using System.Linq;

namespace ProjectStart.Common
{
    public class GenericResponse<T> : Response
    {
        public GenericResponse(T data, string message = null, IDictionary<string, string> errors = null) : base(message, errors)
        {
            Data = data;
        }
        public T Data { get; set; }
    }
    public static class ResponseExtension
    {
        public static GenericResponse<T> ToResponse<T>(this T data, bool success = true, string message = null, IDictionary<string, string> errors = null)
            => new GenericResponse<T>(data, message, errors);
    }
}
